src/main/java/com/exasol/bucketfs/HttpResponseEvaluator.java

```diff
@@ -0,0 +1,44 @@
+package com.exasol.bucketfs;
+
+import static com.exasol.errorreporting.ExaError.messageBuilder;
+import static java.net.HttpURLConnection.*;
+
+import java.net.URI;
+
+/**
+ * This class provides a central place for evaluation of HTTP response and throws appropriate exceptions.
+ */
+public class HttpResponseEvaluator {
+
+    private HttpResponseEvaluator() {
+        // only static usage
+    }
+
+    /**
+     * Evaluates the response
```

Comment:
```suggestion
     * Evaluates the HTTP response code.
     * <p>
     * In case of success does nothing, otherwise throws appropriate exception.
     * </p>
```